front/lib/api/files/upload.ts

```diff
@@ -123,67 +113,18 @@ const resizeAndUploadToFileStorage: PreprocessingFunction = async (
   }
 };
 
-async function createFileTextStream(buffer: Buffer, contentType: string) {
-  if (!isTextExtractionSupportedContentType(contentType)) {
-    throw new Error("unsupported_content_type");
-  }
-
-  const extractionRes = await new TextExtraction(
-    config.getTextExtractionUrl()
-  ).fromBuffer(buffer, contentType);
-
-  if (extractionRes.isErr()) {
-    // We must throw here, stream does not support Result type.
-    throw extractionRes.error;
-  }
-
-  const pages = extractionRes.value;
-
-  const prefix = pagePrefixesPerMimeType[contentType];
-
-  return new Readable({
-    async read() {
-      for (const page of pages) {
-        const pageText = prefix
-          ? `${prefix}: ${page.pageNumber}/${pages.length}\n${page.content}\n\n`
-          : page.content;
-        this.push(pageText);
-      }
-      this.push(null);
-    },
-  });
-}
-
-const extractTextFromFile: PreprocessingFunction = async (
+const extractTextFromFileAndUpload: PreprocessingFunction = async (
   auth: Authenticator,
   file: FileResource
 ) => {
   try {
-    const readStream = file.getReadStream({
-      auth,
-      version: "original",
-    });
-    // Load file in memory.
-    const arrayBuffer = await buffer(readStream);
+    const content = await extractTextFromFile(auth, file);
+    const writeStream = file.getWriteStream({ auth, version: "processed" });
 
-    const writeStream = file.getWriteStream({
-      auth,
-      version: "processed",
-    });
-    const textStream = await createFileTextStream(
-      arrayBuffer,
-      file.contentType
-    );
-
-    await pipeline(
-      textStream,
-      async function* (source) {
-        for await (const chunk of source) {
-          yield chunk;
-        }
-      },
-      writeStream
-    );
+    // Use pipeline with an async generator
+    await pipeline(async function* () {
```

Comment:
The `pipeline` isn't needed if you are not using streams. See my comment from [here](https://github.com/dust-tt/dust/pull/8698/files#diff-5729f3e9495a8c90103d498e6faedba01ee7db8530ce28f52f043f77b04dba10R49).

Reply:
As said, will move back to streams in another PR.